sdks/python/apache_beam/transforms/stats_test.py

```diff
@@ -160,57 +159,13 @@ def test_get_sample_size_from_est_error(self):
     assert beam.ApproximateUnique._get_sample_size_from_est_error(0.05) == 1600
     assert beam.ApproximateUnique._get_sample_size_from_est_error(0.01) == 40000
 
-  @unittest.skip(
-      'Skip it because hash function is not good enough. '
-      'TODO: BEAM-7654')
   def test_approximate_unique_global_by_sample_size(self):
     # test if estimation error with a given sample size is not greater than
     # expected max error.
     sample_size = 16
     max_err = 2 / math.sqrt(sample_size)
-    test_input = [
-        4,
-        34,
-        29,
-        46,
-        80,
-        66,
-        51,
-        81,
-        31,
-        9,
-        26,
-        36,
-        10,
-        41,
-        90,
-        35,
-        33,
-        19,
-        88,
-        86,
-        28,
-        93,
-        38,
-        76,
-        15,
-        87,
-        12,
-        39,
-        84,
-        13,
-        32,
-        49,
-        65,
-        100,
-        16,
-        27,
-        23,
-        30,
-        96,
-        54
-    ]
-
+    test_input = list(range(100))
```

Comment:
A fixed dataset was used because random dataset cannot always pass the test even with retries and introduces flakiness. Does it always pass now?

Reply:
yes, all tests pass in a single run!

Reply:
but I will add a seed to account for randomness in the shuffling of the test data.

Reply:
It’s better to run at least 100 times to test flakiness. I’d recommend to run 1000 times.

Reply:
hmm not sure I understand... i believe that adding a random seed should eliminate all randomness in the generation and shuffling of the test data, such that every test run uses the same data?

Reply:
Ah, I miss understood your above comment. Adding the seed looks good to me.